recipes/stlab/old/conanfile.py

```diff
@@ -0,0 +1,187 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, collect_libs, copy, get, rename, replace_in_file, rmdir, save
+from conan.tools.scm import Version
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+required_conan_version = ">=1.52.0"
+
+class Stlab(ConanFile):
+    name = 'stlab'
+    description = 'The Software Technology Lab libraries.'
+    url = 'https://github.com/conan-io/conan-center-index'
+    homepage = 'https://github.com/stlab/libraries'
+    license = 'BSL-1.0'
+    topics = 'conan', 'c++', 'concurrency', 'futures', 'channels'
+
+    settings = "arch", "os", "compiler", "build_type",
+
+    options = {
+        "boost_optional": [True, False],
+        "boost_variant": [True, False],
+        "coroutines": [True, False],
+        "task_system": ["portable", "libdispatch", "emscripten", "pnacl", "windows", "auto"],
+    }
+
+    default_options = {
+        "boost_optional": False,
+        "boost_variant": False,
+        "coroutines": False,
+        "task_system": "auto",
+    }
+
+    no_copy_source = True
+    _source_subfolder = 'source_subfolder'
+
+    # short_paths = True
+    # def layout(self):
+    #     cmake_layout(self, src_folder="src")
+
+    def _use_boost(self):
+        return self.options.boost_optional or self.options.boost_variant
+
+    def _requires_libdispatch(self):
+        # On macOS it is not necessary to use the libdispatch conan package, because the library is
+        # included in the OS.
+        return self.options.task_system == "libdispatch" and self.settings.os != "Macos"
+
+    def requirements(self):
+        if self._use_boost():
+            self.requires("boost/1.75.0")
+
+        if self._requires_libdispatch():
+            self.requires("libdispatch/5.3.2")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def _fix_boost_components(self):
+        if self.settings.os != "Macos": return
+        if self.settings.compiler != "apple-clang": return
+        if Version(self.settings.compiler.version) >= "12": return
+
+        #
+        # On Apple we have to force the usage of boost.variant, because Apple's implementation of C++17 is not complete.
+        #
+        self.output.info("Apple-Clang versions less than 12 do not correctly support std::optional or std::variant, so we will use boost::optional and boost::variant instead.")
+        self.options.boost_optional = True
+        self.options.boost_variant = True
```

Comment:
There should be in configure and validate 🤔 not sure where this is being called it a bit complicated to follow

Reply:
This is the `old` recipe. It remains untouched. It used to work